common/src/async-queue.ts

```diff
@@ -34,6 +35,15 @@ export function createAsyncQueue<T>() {
             throw ended;
     }
 
+    const peek = () => {
+        if (queued.length) {
```

Comment:
No changes should be necessary in this queue, calling clear or dequeue is sufficient. Catch any errors if you need to..